src/odemis/gui/cont/streams.py

```diff
@@ -2732,6 +2735,27 @@ def addMonochromator(self):
                                   play=False
                                   )
 
+    def addTimeCorrelator(self):
+        """ Create a Time Correlator stream and add to to all compatible viewports """
+
+        main_data = self._main_data_model
+        tc_stream = acqstream.ScannedTemporalSettingsStream(
+            "Time Correlator",
+            main_data.time_correlator,
+            main_data.time_correlator.data,
+            main_data.ebeam,
+        )
+
+        # Create the equivalent MDStream
+        sem_stream = self._tab_data_model.semStream
+        sem_tc_stream = acqstream.SEMTemporalMDStream("SEM Time Correlator",
+                                                  [sem_stream, tc_stream])
+
+        return self._addRepStream(tc_stream, sem_tc_stream,
+                                  axes={},
```

Comment:
You'll need tc-od-filter and tc-filter in the axes.

Reply:
Does this mean I also have to add tc-od-filter and tc-filter to model.__init__ and then call it as
`
axes = {"tc-od-filter": main_data.tc_od_filter, "tc-filter": main_data.tc_filter}
` ?

Reply:
Yes, that should be indeed the right way to do it.